### PR TITLE
Add few bounds, and fix build

### PR DESCRIPTION
**Pull request by phadej**
I noticed that you are seeking for a maintainer.

I can help with that task, at least keep `HaskellNet` working with newer GHC and dependency versions. I have good record with that.

### CODE REVIEW

File: HaskellNet.cabal [hunk 1]
```diff
@@ -46,14 +46,17 @@ Library
     Text.Packrat.Pos
     Text.Packrat.Parse
 
+  other-extensions:
+    MultiWayIf
```
Comment (jtdaugherty) [inline HaskellNet.cabal]: Why was this change part of the commit?

**Comment by phadej** [comment]
It’s more precise/semantic way to disallow older GHCs, than to remember which version introduces MultiWayIf. cabal-install solver takes other-extensions information into account (it asks GHC which extensions are supported)

**Comment by jtdaugherty** [comment]
Okay, thanks for the explanation. I can see how that would enforce the desired behavior, but I don't think it is helpful to communicate to humans about which GHC versions are supported. I think it'd be helpful to document that somewhere explicitly rather than using extensions as a proxy.

**Comment by jtdaugherty** [comment]
To follow up and be more specific: I'd be happy to merge this if you can add some comments to the README and the Cabal file about explicit GHC version requirements introduced by this change.

**Comment by phadej** [comment]
I decline to add any comments to README, as the

https://github.com/jtdaugherty/HaskellNet/commit/2808362b36d2f3b618400790ede43311fd2377e4

is the change which started to require GHC-7.6 instead of (at least, there is no reasonable way to check) GHC-7.0.

Adding `other-extensions` is adding *some* metadata, to help solver if you decide to make a release from current `master`.

If you accept my offer to become a (co-)maintainer, I'd remove `MultiWayIf`, and restore compatibility with older GHCs. And also setup CI, so such breakages would be caught early.

**Comment by jtdaugherty** [comment]
> I decline to add any comments to README, as the 2808362 is the change which started to require GHC-7.6

I want to be clear that while I appreciate that adding `other-extensions` has value, it would still be helpful to update the README and add a comment to the Cabal file in cases like this. We have two parties to help: the solver, and the user. This change helps the solver. That's great! It does not help the user at all, and I think that's worth doing. The commit that introduced the requirement is not a justification for not documenting this now.

On the topic of maintenance, @lemol had previously been involved with that but has not been active for some time as far as I can tell. I have not had time to do the maintenance myself, so I am definitely interested in finding others who have the time and the energy to maintain this. With that said, I am wary of offering the maintenance option to someone who has only submitted one patch. For me to feel confident that you are in a good position to take on maintenance, I think I'd want to see more patches and I'd also want us to be in agreement about the nature of the changes. (For example, I would fully support removing `MultiWayIf` entirely as you mentioned, along with setting up CI, etc.)

**Pull request merged**